### ISSUE TITLE
Upgrade doesn't start

### ISSUE THREAD

**Issue by abullones** [nilsteampassnet/TeamPass#1739]
Hello! I'm triying to upgrade my TeamPass server from 2.1.26 to 2.1.27.  It's installed on a CentOs server.  after following the instructions on the documentation, i go to the upgrade link to begin the process and it asks me for my admin credentials.  Entering the u/p, a loading bar starts moving like on what i guess is a credential's check but it doesn't do anything besides moving one side to the other endlessly.  Could anybody think of what i could be missing?
![captura de pantalla 2017-04-24 a las 16 43 31](https://cloud.githubusercontent.com/assets/2195521/25342744/33e5020c-290d-11e7-97e9-c688d37293c3.png)

**Comment by abullones** [comment]
I add an entry i found on the error log 
[Mon Apr 24 16:47:45.450838 2017] [:error] [pid 1834] [client **.*.***.**:*****] PHP Fatal error:  Uncaught exception 'configFileNotFoundException' with message 'OWASP CSRFProtector: configuration file not found for CSRFProtector!' in /var/www/html/teampass/includes/libraries/csrfp/libs/csrf/csrfprotector.php:123\nStack trace:\n#0 /var/www/html/teampass/index.php(45): csrfProtector::init()\n#1 {main}\n  thrown in /var/www/html/teampass/includes/libraries/csrfp/libs/csrf/csrfprotector.php on line 123, referer: http://***.***.***.****/teampass/index.php?page=manage_settings

and this on the access log:88.2.252.45 - - [24/Apr/2017:16:47:45 +0200] "GET /teampass/index.php?page=manage_folders HTTP/1.1" 500 - "http://***.***.***.***/teampass/index.php?page=manage_settings" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36"
88.2.252.45 - - [24/Apr/2017:16:48:32 +0200] "-" 408 - "-" "-"
88.2.252.45 - - [24/Apr/2017:16:48:32 +0200] "-" 408 - "-" "-"
88.2.252.45 - - [24/Apr/2017:16:48:32 +0200] "-" 408 - "-" "-"
88.2.252.45 - - [24/Apr/2017:16:48:35 +0200] "-" 408 - "-" "-

**Comment by nilsteampassnet** [comment]
Are you sure to have copied the file `includes\libraries\csrfp\libs\csrfp.config.php` from your previous Teampass instance to the new one?

**Comment by abullones** [comment]
Hi there, thanks for helping.  I followed this doc to upgrade: https://teampass.readthedocs.io/en/latest/install/upgrade/ and i don't see mentioned to copy csrfp.config.php.  Following your advice, i copied it and tried again with the same results but now the logs are different:
access_log 
xxx.xxx.xxx.xxx - - [25/Apr/2017:10:36:28 +0200] "GET /teampass/install/upgrade.php HTTP/1.1" 200 14194 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36"
xxx.xxx.xxx.xxx - - [25/Apr/2017:10:36:35 +0200] "POST /teampass/install/upgrade_ajax.php HTTP/1.1" 500 - "http://xxx.xxx.xxx.xxx/teampass/install/upgrade.php" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36"

error_log
[Tue Apr 25 10:36:15.537893 2017] [suexec:notice] [pid 828] AH01232: suEXEC mechanism enabled (wrapper: /usr/sbin/suexec)
[Tue Apr 25 10:36:15.621493 2017] [auth_digest:notice] [pid 828] AH01757: generating secret for digest authentication ...
[Tue Apr 25 10:36:15.621518 2017] [lbmethod_heartbeat:notice] [pid 828] AH02282: No slotmem from mod_heartmonitor
[Tue Apr 25 10:36:15.739743 2017] [mpm_prefork:notice] [pid 828] AH00163: Apache/2.4.6 (CentOS) PHP/5.6.30 configured -- resuming normal operations
[Tue Apr 25 10:36:15.739755 2017] [core:notice] [pid 828] AH00094: Command line: '/usr/sbin/httpd -D FOREGROUND'

I'm not able to identify an error now.  Besides I doubting if I'm following the correct procedure to upgrade the server.  Could you please tell me what you think?  Thanks!

**Comment by albernieto** [comment]
I have this problem in the upgrade with the same logs. The upgrade never start!!

**Comment by afarrapeira** [comment]
Same here, I'm getting an error upgrading from 2.1.26.15 to 2.1.27.3

I'm getting a `NetworkError: 500 Internal Server Error - http://$host/teampass/install/upgrade_ajax.php` in the first page of the upgrade, just after typing the administrator login and password.

**Comment by nilsteampassnet** [comment]
I've tried with the same version of PHP and could  not reproduced.
The errors you show are not really errors from my point of view.
Did you tried to increase the level of errors displayed by Apache?
Are you all using a Mac?

@afarrapeira what are the associated logs error you have?

**Comment by afarrapeira** [comment]
My server configuration is:

**Operating system:**
Ubuntu Server 14.04 64-bit

**Web server:**
Apache/2.4.23 (Ubuntu)

**Database:**
5.7.12-1

**PHP version:**
PHP 7.0.9-1

I don't know what happened yesterday... After getting that error I restored the 2.1.26 version I had in "teampass_old" after moving the new version to "teampass_27".
Today I tried it again: I renamed back "teampass" to "teampass_old", and "teampass_27" to "teampass", restarted Apache just in case and the upgrade worked like a charm.

I hope you get to fix your issues soon too, @abullones and @albernieto .

@nilsteampassnet other than the Error 500 in the access log I wasn't getting anything weird in the error log, it looked mostly like [abullones](#issuecomment-296962156) error log. I increased Apache's LogLevel today, but it already worked so I couldn't find anything :neutral_face:

**Comment by abullones** [comment]
Hi, i do have a mac, i just tried it on windows with the same results

**Comment by albernieto** [comment]
My server configuration is:

Operating system:
Debian 9

Web server:
Apache/2.4.25-3

Database:
mysql-mariadb

PHP version:
PHP 7.0.16-3

After copy the file includes\libraries\csrfp\libs\csrfp.config.php the error is diferent but the result is the same: the upgrade not start

Apache logging debug:

Access.log
10.69.100.10 - - [05/May/2017:12:35:37 +0200] "GET /teampass/install/upgrade.php HTTP/1.1" 200 3837 "-" "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"
10.69.100.10 - - [05/May/2017:12:35:38 +0200] "GET /browserconfig.xml HTTP/1.1" 404 509 "-" "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"
10.69.100.10 - - [05/May/2017:12:35:55 +0200] "POST /teampass/install/upgrade_ajax.php HTTP/1.1" 500 295 "http://10.69.100.5/teampass/install/upgrade.php" "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"

Error.log 
[Fri May 05 12:35:37.537148 2017] [authz_core:debug] [pid 724] mod_authz_core.c(809): [client 10.69.100.10:54595] AH01626: authorization result of Require all granted: granted
[Fri May 05 12:35:37.537230 2017] [authz_core:debug] [pid 724] mod_authz_core.c(809): [client 10.69.100.10:54595] AH01626: authorization result of <RequireAny>: granted
[Fri May 05 12:35:37.931739 2017] [deflate:debug] [pid 724] mod_deflate.c(853): [client 10.69.100.10:54595] AH01384: Zlib: Compressed 14194 to 3266 : URL /teampass/install/upgrade.php
[Fri May 05 12:35:38.330628 2017] [watchdog:debug] [pid 765] mod_watchdog.c(563): AH02980: Watchdog: nothing configured?
[Fri May 05 12:35:38.882598 2017] [authz_core:debug] [pid 725] mod_authz_core.c(809): [client 10.69.100.10:54596] AH01626: authorization result of Require all granted: granted
[Fri May 05 12:35:38.882727 2017] [authz_core:debug] [pid 725] mod_authz_core.c(809): [client 10.69.100.10:54596] AH01626: authorization result of <RequireAny>: granted
[Fri May 05 12:35:38.882975 2017] [core:info] [pid 725] [client 10.69.100.10:54596] AH00128: File does not exist: /var/www/html/browserconfig.xml
[Fri May 05 12:35:39.338453 2017] [watchdog:debug] [pid 767] mod_watchdog.c(563): AH02980: Watchdog: nothing configured?
[Fri May 05 12:35:39.338801 2017] [watchdog:debug] [pid 766] mod_watchdog.c(563): AH02980: Watchdog: nothing configured?
[Fri May 05 12:35:55.811629 2017] [authz_core:debug] [pid 722] mod_authz_core.c(809): [client 10.69.100.10:54601] AH01626: authorization result of Require all granted: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Fri May 05 12:35:55.811790 2017] [authz_core:debug] [pid 722] mod_authz_core.c(809): [client 10.69.100.10:54601] AH01626: authorization result of <RequireAny>: granted, referer: http://10.69.100.5/teampass/install/upgrade.php

**Comment by nilsteampassnet** [comment]
Do you all have mod_watchdog installed?

Once file `csrfp.config.php` copied to correct folder, do you manage to move forward after admin identification?

**Comment by albernieto** [comment]
In the folder mods-enabled and mods-avaliable the mods_watchdog don´t exist

After copy the csrfp.confg.php put muy admin user (teampass_admin) and passwod but the upgrade don´t start. 

Today:

error.log

[Mon May 08 12:59:33.977030 2017] [authz_core:debug] [pid 9325] mod_authz_core.c(809): [client 10.69.100.10:55873] AH01626: authorization result of Require all granted: granted
[Mon May 08 12:59:33.977176 2017] [authz_core:debug] [pid 9325] mod_authz_core.c(809): [client 10.69.100.10:55873] AH01626: authorization result of <RequireAny>: granted
[Mon May 08 12:59:33.993852 2017] [deflate:debug] [pid 9325] mod_deflate.c(853): [client 10.69.100.10:55873] AH01384: Zlib: Compressed 14194 to 3266 : URL /teampass/install/upgrade.php
[Mon May 08 12:59:34.706143 2017] [authz_core:debug] [pid 9325] mod_authz_core.c(809): [client 10.69.100.10:55873] AH01626: authorization result of Require all granted: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.706193 2017] [authz_core:debug] [pid 9325] mod_authz_core.c(809): [client 10.69.100.10:55873] AH01626: authorization result of <RequireAny>: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.723553 2017] [deflate:debug] [pid 9325] mod_deflate.c(853): [client 10.69.100.10:55873] AH01384: Zlib: Compressed 31000 to 7035 : URL /teampass/includes/font-awesome/css/font-awesome.min.css, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.784987 2017] [authz_core:debug] [pid 9322] mod_authz_core.c(809): [client 10.69.100.10:55877] AH01626: authorization result of Require all granted: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.785128 2017] [authz_core:debug] [pid 9322] mod_authz_core.c(809): [client 10.69.100.10:55877] AH01626: authorization result of <RequireAny>: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.794549 2017] [authz_core:debug] [pid 9323] mod_authz_core.c(809): [client 10.69.100.10:55880] AH01626: authorization result of Require all granted: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.794653 2017] [authz_core:debug] [pid 9323] mod_authz_core.c(809): [client 10.69.100.10:55880] AH01626: authorization result of <RequireAny>: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.799711 2017] [deflate:debug] [pid 9322] mod_deflate.c(853): [client 10.69.100.10:55877] AH01384: Zlib: Compressed 1181 to 479 : URL /teampass/install/install.css, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.808000 2017] [authz_core:debug] [pid 9324] mod_authz_core.c(809): [client 10.69.100.10:55879] AH01626: authorization result of Require all granted: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.808102 2017] [authz_core:debug] [pid 9324] mod_authz_core.c(809): [client 10.69.100.10:55879] AH01626: authorization result of <RequireAny>: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.808652 2017] [authz_core:debug] [pid 9321] mod_authz_core.c(809): [client 10.69.100.10:55878] AH01626: authorization result of Require all granted: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.808731 2017] [authz_core:debug] [pid 9321] mod_authz_core.c(809): [client 10.69.100.10:55878] AH01626: authorization result of <RequireAny>: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.809394 2017] [authz_core:debug] [pid 9323] mod_authz_core.c(809): [client 10.69.100.10:55880] AH01626: authorization result of Require all granted: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.809414 2017] [authz_core:debug] [pid 9323] mod_authz_core.c(809): [client 10.69.100.10:55880] AH01626: authorization result of <RequireAny>: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.809738 2017] [authz_core:debug] [pid 9322] mod_authz_core.c(809): [client 10.69.100.10:55877] AH01626: authorization result of Require all granted: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.809757 2017] [authz_core:debug] [pid 9322] mod_authz_core.c(809): [client 10.69.100.10:55877] AH01626: authorization result of <RequireAny>: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.809951 2017] [authz_core:debug] [pid 9325] mod_authz_core.c(809): [client 10.69.100.10:55873] AH01626: authorization result of Require all granted: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.809970 2017] [authz_core:debug] [pid 9325] mod_authz_core.c(809): [client 10.69.100.10:55873] AH01626: authorization result of <RequireAny>: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.818264 2017] [deflate:debug] [pid 9321] mod_deflate.c(853): [client 10.69.100.10:55878] AH01384: Zlib: Compressed 19535 to 5899 : URL /teampass/install/js/aes.min.js, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.831931 2017] [deflate:debug] [pid 9322] mod_deflate.c(853): [client 10.69.100.10:55877] AH01384: Zlib: Compressed 91555 to 32040 : URL /teampass/install/js/jquery.min.js, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.832614 2017] [watchdog:debug] [pid 10413] mod_watchdog.c(563): AH02980: Watchdog: nothing configured?
[Mon May 08 12:59:34.833154 2017] [authz_core:debug] [pid 10413] mod_authz_core.c(809): [client 10.69.100.10:55881] AH01626: authorization result of Require all granted: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.833173 2017] [authz_core:debug] [pid 10413] mod_authz_core.c(809): [client 10.69.100.10:55881] AH01626: authorization result of <RequireAny>: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.834542 2017] [deflate:debug] [pid 10413] mod_deflate.c(853): [client 10.69.100.10:55881] AH01384: Zlib: Compressed 3033 to 1252 : URL /teampass/install/upgrade.js, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.844124 2017] [authz_core:debug] [pid 9325] mod_authz_core.c(809): [client 10.69.100.10:55873] AH01626: authorization result of Require all granted: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.844159 2017] [authz_core:debug] [pid 9325] mod_authz_core.c(809): [client 10.69.100.10:55873] AH01626: authorization result of <RequireAny>: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.850856 2017] [deflate:debug] [pid 9323] mod_deflate.c(853): [client 10.69.100.10:55880] AH01384: Zlib: Compressed 187815 to 48160 : URL /teampass/install/js/jquery-ui.min.js, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:34.858276 2017] [deflate:debug] [pid 9325] mod_deflate.c(853): [client 10.69.100.10:55873] AH01384: Zlib: Compressed 8505 to 3004 : URL /teampass/includes/js/functions.js, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:35.164674 2017] [authz_core:debug] [pid 9324] mod_authz_core.c(809): [client 10.69.100.10:55879] AH01626: authorization result of Require all granted: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:35.164732 2017] [authz_core:debug] [pid 9324] mod_authz_core.c(809): [client 10.69.100.10:55879] AH01626: authorization result of <RequireAny>: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:35.791175 2017] [authz_core:debug] [pid 9321] mod_authz_core.c(809): [client 10.69.100.10:55878] AH01626: authorization result of Require all granted: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:35.791217 2017] [authz_core:debug] [pid 9321] mod_authz_core.c(809): [client 10.69.100.10:55878] AH01626: authorization result of <RequireAny>: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:35.831071 2017] [watchdog:debug] [pid 10414] mod_watchdog.c(563): AH02980: Watchdog: nothing configured?
[Mon May 08 12:59:36.836706 2017] [watchdog:debug] [pid 10415] mod_watchdog.c(563): AH02980: Watchdog: nothing configured?
[Mon May 08 12:59:37.841381 2017] [watchdog:debug] [pid 10416] mod_watchdog.c(563): AH02980: Watchdog: nothing configured?
[Mon May 08 12:59:38.846301 2017] [watchdog:debug] [pid 10417] mod_watchdog.c(563): AH02980: Watchdog: nothing configured?
[Mon May 08 12:59:39.853414 2017] [watchdog:debug] [pid 10418] mod_watchdog.c(563): AH02980: Watchdog: nothing configured?
[Mon May 08 12:59:55.427062 2017] [authz_core:debug] [pid 10415] mod_authz_core.c(809): [client 10.69.100.10:55929] AH01626: authorization result of Require all granted: granted, referer: http://10.69.100.5/teampass/install/upgrade.php
[Mon May 08 12:59:55.427126 2017] [authz_core:debug] [pid 10415] mod_authz_core.c(809): [client 10.69.100.10:55929] AH01626: authorization result of <RequireAny>: granted, referer: http://10.69.100.5/teampass/install/upgrade.php

access.log

10.69.100.10 - - [08/May/2017:12:59:33 +0200] "GET /teampass/install/upgrade.php HTTP/1.1" 200 3839 "-" "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"
10.69.100.10 - - [08/May/2017:12:59:34 +0200] "GET /teampass/includes/font-awesome/css/font-awesome.min.css HTTP/1.1" 200 7390 "http://10.69.100.5/teampass/install/upgrade.php" "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"
10.69.100.10 - - [08/May/2017:12:59:34 +0200] "GET /teampass/includes/images/canevas/logo.png HTTP/1.1" 304 182 "http://10.69.100.5/teampass/install/upgrade.php" "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"
10.69.100.10 - - [08/May/2017:12:59:34 +0200] "GET /teampass/install/install.css HTTP/1.1" 200 833 "http://10.69.100.5/teampass/install/upgrade.php" "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"
10.69.100.10 - - [08/May/2017:12:59:34 +0200] "GET /teampass/install/images/76.gif HTTP/1.1" 304 182 "http://10.69.100.5/teampass/install/upgrade.php" "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"
10.69.100.10 - - [08/May/2017:12:59:34 +0200] "GET /teampass/install/images/error.png HTTP/1.1" 304 180 "http://10.69.100.5/teampass/install/upgrade.php" "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"
10.69.100.10 - - [08/May/2017:12:59:34 +0200] "GET /teampass/install/js/aes.min.js HTTP/1.1" 200 6269 "http://10.69.100.5/teampass/install/upgrade.php" "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"
10.69.100.10 - - [08/May/2017:12:59:34 +0200] "GET /teampass/install/js/jquery.min.js HTTP/1.1" 200 32411 "http://10.69.100.5/teampass/install/upgrade.php" "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"
10.69.100.10 - - [08/May/2017:12:59:34 +0200] "GET /teampass/install/upgrade.js HTTP/1.1" 200 1621 "http://10.69.100.5/teampass/install/upgrade.php" "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"
10.69.100.10 - - [08/May/2017:12:59:34 +0200] "GET /teampass/install/js/jquery-ui.min.js HTTP/1.1" 200 48531 "http://10.69.100.5/teampass/install/upgrade.php" "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"
10.69.100.10 - - [08/May/2017:12:59:34 +0200] "GET /teampass/includes/js/functions.js HTTP/1.1" 200 3373 "http://10.69.100.5/teampass/install/upgrade.php" "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"
10.69.100.10 - - [08/May/2017:12:59:35 +0200] "GET /teampass/includes/images/canevas/bg_home.jpg HTTP/1.1" 304 180 "http://10.69.100.5/teampass/install/upgrade.php" "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"
10.69.100.10 - - [08/May/2017:12:59:35 +0200] "GET /teampass/includes/images/canevas/footer_all.png HTTP/1.1" 304 180 "http://10.69.100.5/teampass/install/upgrade.php" "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"
::1 - - [08/May/2017:12:59:42 +0200] "OPTIONS * HTTP/1.0" 200 126 "-" "Apache/2.4.25 (Debian) (internal dummy connection)"
10.69.100.10 - - [08/May/2017:12:59:55 +0200] "POST /teampass/install/upgrade_ajax.php HTTP/1.1" 500 295 "http://10.69.100.5/teampass/install/upgrade.php" "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"

![captura](https://cloud.githubusercontent.com/assets/24434136/25801614/af521f56-33ee-11e7-900b-f33cd76628aa.JPG)

**Comment by dataegg** [comment]
Hi There,

I also faced a problem like this but I did not find any errors in server logging. I resolved it by restarting the upgrade from scratch and copied csrfp.config.php to the new installation ( i didn't do this the first time). I had a look in csrfp.config.php and for me there was a wrong value for "jsUrl", it pointed to my production environment and I'm upgrading my test environment. I probably copied this file some time before while refreshing the test environment.

Hope this helps someone

**Comment by romainlombart** [comment]
![upgrade_dont_start](https://cloud.githubusercontent.com/assets/10233399/25992507/029ee0e8-3707-11e7-86da-8402a06ec6a4.PNG)

Hello,
Same here, upgrade doesn't start from 2.1.26 to 2.1.27.3

My Config :
Synology web station 2.0.1-0111
Apache 2.2.31-0017
Maria-db 5.5.54-0076
PHP 7.0.16-0021 or PHP 5.6.30-0052 (Same problem)

I followed the latest doc upgrade :
http://teampass.readthedocs.io/en/latest/install/upgrade/

**Comment by romainlombart** [comment]
Same issue with 2.1.27.4 from 2.1.26
Is there another way to upgrade the database?

**Comment by romainlombart** [comment]
It's good I found the problem after quite a lot of tests :

On the file /includes/config/settings.php the path of theses lines was not correct, I modified it to match my installation :
`@date_default_timezone_set($_SESSION['settings']['timezone']);
@define('SECUREPATH', '/volume1/web/teampass/includes');
require_once "/volume1/web/teampass/includes/sk.php";
@define('COST', '13'); // Don't change this.`

I was obliged to also copy the following file for the update to work :
/includes/sk.php

**Comment by nilsteampassnet** [comment]
Thank you for this return of experience.
I could reproduce too and committed a fix that checks this on upgrade start.

**Comment by srvMelchior** [comment]
I had the same problem upgrading from 2.1.27.10 to 2.1.27.11.  We found the solution was a restart of the apache service and clear its cache.
Our setup has Tempass not installed in a generic "Teampass" folder, but one labeled like the zip i.e. "/TeamPass-2.1.27.10". The path used by Teampass refers to a symbolic link (called "/teampass") that is on the same level and then points to the actual directory. So our process when upgrading was to first make a copy of the 27.10 folder, then copy the content of the 27.11 into this new folder, overwriting the old files, call this folder 27.11 and set the link to now point to the 27.11 directory. We found that the apache error log still expected the old path. A restart solved the problem.

I hope this helps some people who find this thread in the future :)
